Add com.github.lxgr_linux.pokete

MaFeLP opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/lxgr-linux/pokete/pull/195
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,72 @@
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/pokete
```
```suggestion
command: pokete
```

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,72 @@
+      # Copy Files
+      - "cp -r . /app"
+      # Create the desktop entry
+      - "sed -i 's#Icon=/usr/share/pokete/assets/pokete.png#Icon=com.github.lxgr_linux.pokete#g' assets/pokete.desktop"
+      - "sed -i 's#Exec=/usr/bin/pokete.py#Exec=/app/bin/pokete#g' assets/pokete.desktop"
```
1. these are very wrong upstream. Notably `Icon` is *always* a name, never a path
2. the `Exec` should be just the command name, not an absolute path
3. `desktop-file-edit` is preferred over obscure sed commands, as it is more robust.

```suggestion
      - desktop-file-edit --set-icon=$FLATPAK_ID --set-key=Exec --set-value=pokete assets/pokete.desktop
```
(this should work regardless 1 and 2 being fixed later)

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,72 @@
+      - "sed -i 's#Exec=/usr/bin/pokete.py#Exec=/app/bin/pokete#g' assets/pokete.desktop"
+      - "install -D assets/pokete.desktop /app/share/applications/com.github.lxgr_linux.pokete.desktop"
+      # Note: The .png icon is installed which does not work due to incorrect size. A patch has been submitted upstream and is waiting for release.
+      - "install -D assets/pokete.png /app/share/icons/hicolor/296x256/com.github.lxgr_linux.pokete.png"
```
`296x256` is an invalid icon size.

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,72 @@
+  # Store save files and logs
+  - "--filesystem=xdg-cache:create"
+  # Trading and update checking
+  - "--share=network"
```
nothing to display?

MaFeLP replied:
No. This is a TUI game.

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,72 @@
+        url: "https://github.com/lxgr-linux/pokete/archive/refs/tags/0.7.2.tar.gz"
+        sha256: "7410b2043a53e636bfb8c9ede4c24f933f2bd3f5685ff0b2b3031206d7cbd5f4"
+
+cleanup:
```
the global cleanup section is usually above `modules`

hfiguiere commented:
bot, build com.github.lxgr_linux.pokete

hfiguiere commented:
there is no AppStream file installed.

MaFeLP commented:
Hi!

Thanks for reviewing this PR!

I fixed the icon and added an AppStream file.

Would it be possible to give @lxgr-linux writing permissions to the new repository as well, once it has been created?

Thank you in advance!

hfiguiere commented:
bot, build com.github.lxgr_linux.pokete

hfiguiere commented:
```
builddir/files/share/appdata/com.github.lxgr_linux.pokete.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
Validation of files failed
```

MaFeLP commented:
I updated the AppStream file upstream, and it now passes my local validation tests.

nedrichards commented:
bot, build com.github.lxgr_linux.pokete

nedrichards commented:
Hrm, crash on startup with:

> Traceback (most recent call last):
  File "/app/pokete.py", line 1411, in <module>
    logging.basicConfig(filename=log_file,
  File "/usr/lib/python3.9/logging/__init__.py", line 2003, in basicConfig
    h = FileHandler(filename, mode,
  File "/usr/lib/python3.9/logging/__init__.py", line 1146, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/lib/python3.9/logging/__init__.py", line 1175, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding,
FileNotFoundError: [Errno 2] No such file or directory: '/var/home/$username/.cache/pokete/pokete.log'

MaFeLP commented:
I had the wrong file system permissions configured. It should work now.

nedrichards commented:
bot, build com.github.lxgr_linux.pokete

hfiguiere commented on com.github.lxgr_linux.pokete.yml:
```diff
@@ -0,0 +1,69 @@
+  - "/Pipfile*"
+
+modules:
+  # Dependency of scrap engine; generated with flatpak-pip-generator
```
just a trick: we usually *include* the json generated by flatpak-pip-generator for ease of maintainance: you can just regenerate the file

(It's not blocking, but this is a common practice)

nedrichards commented:
/merge @lxgr-linux
